### PR TITLE
Refactor SqlTable

### PR DESCRIPTION
**Pull request by astaric**
Sql code is currently coupled with SqlTable, making support for other sql backends hard to implement. This PR moves (some of the) code that is dependent on sql driver to a separate class (Psycopg2Backend) which could (in theory) be replaces by another implementation of the database backend.

### CODE REVIEW

**Comment by astaric** [comment]
@lanzagar I have added some tests for the sql calls that were not covered before and added docstrings for the public Backend interface. Please have another look.

File: Orange/widgets/data/owsql.py [hunk 1]
```diff
@@ -309,10 +309,10 @@ def get_table(self):
             QApplication.setOverrideCursor(QCursor(Qt.WaitCursor))
             if sample:
                 s = table.sample_time(1)
-                domain = s.get_domain(guess_values=True)
+                domain = s.create_domain(inspect_values=True)
```
Comment (lanzagar) [inline Orange/widgets/data/owsql.py]: The new create domain has a required positional argument

File: Orange/tests/sql/test_misc.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+"""Test for miscelaneous sql queries in widgets
+
+Please not that such use is deprecated.
```
Comment (lanzagar) [inline Orange/tests/sql/test_misc.py]: note

File: Orange/data/sql/table.py [hunk 3]
```diff
@@ -75,53 +289,28 @@ def __init__(
         """
         if isinstance(connection_params, str):
             connection_params = dict(database=connection_params)
-        self.connection_params = connection_params
```
Comment (lanzagar) [inline Orange/data/sql/table.py]: was used in (at least) owdatainfo.py, L142

**Pull request merged**